ner_v1/detectors/textual/text/text_detection.py

```diff
@@ -313,61 +350,68 @@ def _text_detection_with_variants(self):
-        return value_final_list, original_final_list
-
-    def _get_entity_substring_from_text(self, variant):
+
+        original_final_list_ = []
+        value_final_list_ = []
+        text = [u' '.join(TOKENIZER.tokenize(processed_text)) for processed_text in self.processed_text]
+
+        _variants_to_values_list = self.db.get_similar_dictionary(entity_name=self.entity_name,
+                                                                  text=text,
+                                                                  fuzziness_threshold=self._fuzziness,
+                                                                  search_language_script=self._target_language_script)
+        for index, _variants_to_values in enumerate(_variants_to_values_list):
+            original_final_list = []
+            value_final_list = []
+            variants_to_values = collections.OrderedDict()
+            for variant, value in iteritems(_variants_to_values):
+                variant = variant.lower()
+                if isinstance(variant, bytes):
+                    variant = variant.decode('utf-8')
+
+                variants_to_values[variant] = value
+            variants_list = variants_to_values.keys()
+
+            # Length based ordering, this reorders the results from datastore
+            # that are already sorted by some relevance scoring
+
+            exact_matches, fuzzy_variants = [], []
+            _text = text
+            for variant in variants_list:
+                if u' '.join(TOKENIZER.tokenize(variant)) in _text[index]:
+                    exact_matches.append(variant)
+                else:
+                    fuzzy_variants.append(variant)
+            exact_matches.sort(key=lambda s: len(TOKENIZER.tokenize(s)), reverse=True)
+            fuzzy_variants.sort(key=lambda s: len(TOKENIZER.tokenize(s)), reverse=True)
+            variants_list = exact_matches + fuzzy_variants
+
+            for variant in variants_list:
+
+                original_text = self._get_entity_substring_from_text(variant, self.processed_text[index])
+                if original_text:
+                    value_final_list.append(variants_to_values[variant])
+                    original_final_list.append(original_text)
+                    _pattern = re.compile(r'\b%s\b' % re.escape(original_text), re.UNICODE)
+                    self.tagged_text[index] = _pattern.sub(self.tag, self.tagged_text[index])
+                    # Instead of dropping completely like in other entities,
+                    # we replace with tag to avoid matching non contiguous segments
+                    self.processed_text[index] = _pattern.sub(self.tag, self.processed_text[index])
+            value_final_list_.append(value_final_list)
+            original_final_list_.append(original_final_list)
+
+        return value_final_list_, original_final_list_
+
+    def _get_entity_substring_from_text(self, variant, sentence):
```

Comment:
Change to `_get_entity_substring_from_text(self, text, variant)`

Reply:
Done